# Information Systems - Midterm Exam (Fall 2025)

## Part II - Computer Portion 
This part is worth 40 points.

### Section A - Data Analysis in Jupyter with pandas
This section is worth 14 points.  

See the Canvas page for:
- **Materials you may use / not use / collaboration policy**
- submission instructions  

For this part of the exam you will be working with a dataset listing CGA cadet activities. The data is available on Canvas from the same page as this notebook. **You are to write Python & pandas code to answer the numbered questions below.** Note most of the questions can be completed independently of one another.

We start by loading the pandas library for you, and supplying the code to import the data stored in the csv as a pandas dataframe named **df**.

In [1]:
import pandas as pd
df = pd.read_csv('c_a.csv')
df

,ActivityName,ActivityType,Ath. Ap,Verified Dt,RosterCount
0,Activity Soccer Club,Cadet Activity,False,8/20/2025 9:50:05 AM,71
1,Adopted Students Association,Cadet Activity,False,9/30/2025 9:18:38 AM,220
2,Adventure Sports,Cadet Activity,False,9/1/2025 9:34:16 PM,348
3,Alfa Company,Cadet Company,False,10/21/2025 9:02:00 AM,143
4,Alpine Race Team,Cadet Activity,False,10/10/2025 1:02:56 PM,82
...,...,...,...,...,...
127,Women's Varsity Basketball,IC Athletics,True,10/8/2025 10:53:26 AM,13
128,Women's Water Polo,Cadet Activity,False,9/3/2025 1:14:21 PM,51
129,Women's Wrestling,Cadet Activity,False,8/27/2025 3:30:19 PM,40
130,Wrestling,IC Athletics,True,10/2/2025 2:13:11 PM,54


1. Assuming there are 1100 cadets at the US Coast Guard Academy, create a new column in the data frame named "Pct of Core" that shows what percentage of the core is on the club roster. For example, for the activity "Adopted Students Association" with 220 members, the "Pct of Core" value would be 0.2 (since that's 220/1100).  **Display the resulting data frame to show the new column.**

In [ ]:
""" YOUR CODE HERE """


2. Find the average roster count over all of the clubs.

In [ ]:
""" YOUR CODE HERE"""


3. Create a data frame that only contains clubs whose activity type is 'Club Sport' or 'IC Athletics'. Then compute the number of rows in it. **Print the data frame and how many rows it has.**  (Hint: you can use the | character to "OR" 2 conditions together.)

In [ ]:
""" YOUR CODE HERE """


4. Returning to the original data frame, use the appropriate pandas command to group the data frame by "ActivityType". Then issue a pandas command to show the data frame organized into these groups. (Hint: There is a one-line way to do the latter that we covered that would result in a dictionary keyed by "ActivityType" with values being lists of row numbers.) 

In [ ]:
""" YOUR CODE HERE """


5. Create a bar chart that shows how many clubs there are for each activity type. (Hint: using the groups created in the previous question, this can be achieved in one line.)

In [ ]:
""" YOUR CODE HERE """
